src/core/activityContent/validation.ts

```diff
@@ -40,386 +42,388 @@ const SUPPORTED_IMAGE_MEDIA_TYPES = [
 ];
 const SUPPORTED_VIDEO_MEDIA_TYPES = ["video/mpeg", "video/ogg", "video/webm", "video/H256", "video/mp4"];
 
-const isActivityContentAttachment = (obj: unknown): obj is ActivityContentAttachment =>
-  isActivityContentAudio(obj) ||
-  isActivityContentImage(obj) ||
-  isActivityContentVideo(obj) ||
-  isActivityContentLink(obj);
-
-const isActivityContentAudioLink = (obj: unknown): obj is ActivityContentAudioLink => {
-  if (!isRecord(obj)) return false;
-  if (obj["type"] !== "Link") return false;
-  if (!isString(obj["href"])) return false;
-  if (!isString(obj["mediaType"])) return false;
-  if (!isArrayOfType(obj["hash"], isActivityContentHash)) return false;
-
-  return true;
+const InvalidRecordMessage = (extra?: string): string => {
+  return `${extra || "object"} is not a record type`;
+};
+const requireIsString = (obj: unknown, name: string): string => {
```

Comment:
These need to return the object because they're used to guarantee the type returned in order to be passed to other functions. I'll try to come up with a better name.

Reply:
A type guard function can also guarantee the type: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types